### ISSUE TITLE
Fusilli BatchNorm compile failure when bumping to `3.12.0rc20260402`

### ISSUE THREAD

**Issue by IanWood1** [iree-org/iree#24008]
### What happened?

Error seen on https://github.com/iree-org/fusilli/pull/293

Compiling `torch.aten.native_batch_norm` with `training=true` targeting the ROCM backend fails with `'func.func' op failed to distribute` when split-reduction is enabled. Bisect points to https://github.com/iree-org/iree/pull/23916 @bangtianliu 

### Steps to reproduce your issue

Three MLIR files, each triggering the same failure with different shapes and
scale/bias configurations:
- [`batchnorm_train_asm_emitter_nchw.mlir`](https://gist.github.com/IanWood1/a5d560f81b7fa1f80a55237d1eb0f2e5)
- [`batchnorm_train_nchw.mlir`](https://gist.github.com/IanWood1/754dd0fb89e4a938c23240999e07e579)
- [`batchnorm_train_nchw_scale_bias.mlir`](https://gist.github.com/IanWood1/fe07a937040bee0a9df7114b5cf75a18)

Command to reproduce (any of the three `.mlir` files):
```bash
iree-compile batchnorm_train_nchw_scale_bias.mlir \
  --iree-hal-target-backends=rocm \
  --iree-rocm-target=mi325x \
  --iree-opt-level=O3 \
  --iree-preprocessing-pass-pipeline='builtin.module(util.func(iree-preprocessing-convert-conv-filter-to-channels-last))' \
  --iree-flow-enable-pad-handling \
  --iree-global-opt-propagate-transposes-through-conv \
  --iree-global-opt-enable-sink-transpose-through-pad \
  --iree-dispatch-creation-enable-fuse-padding-into-linalg-consumer-ops \
  --iree-dispatch-creation-enable-aggressive-reshape-movement \
  --iree-dispatch-creation-enable-split-reduction \
  --iree-torch-externalize-transients \
  --iree-torch-enable-shape-refinement \
  -o /dev/null
```

Or a simplified version that also seems to reproduce the issue:
```
iree-compile repro.mlir \
  --iree-hal-target-backends=rocm \
  --iree-rocm-target=mi325x \
  --iree-opt-level=O3 \
  --iree-dispatch-creation-enable-split-reduction
```

### What component(s) does this issue relate to?

Compiler

### Version information

[`e89e210855`](https://github.com/iree-org/iree/commit/e89e210855a45df61e3088dfb75aed38e82541a0)

**Comment by IanWood1** [comment]
Here's the logs from `batchnorm_train_nchw_scale_bias.mlir` with `--mlir-print-ir-after-all`: https://gist.github.com/IanWood1/e62ad2652668a9b696fabef4428f8fde

**Comment by sommerlukas** [comment]
I've investigated the failures locally; they should be fixed by https://github.com/iree-org/iree/pull/23991. 

I can successfully compile all three reproducers from above locally by applying the changes from https://github.com/iree-org/iree/pull/23991 on top of `main`. @IanWood1 can you check if this also works for you?

**Comment by Groverkss** [comment]
Should be fixed by https://github.com/iree-org/iree/pull/23991
